health/health.c

```diff
@@ -655,6 +655,35 @@ static int update_disabled_silenced(RRDHOST *host, RRDCALC *rc) {
         return 0;
 }
 
+// Create alarms for dimensions that have been added to charts
+// since the previous iteration.
+static void init_pending_foreach_alarms(RRDHOST *host) {
+    rrdhost_wrlock(host);
+
+    if (host->alarms_with_foreach || host->alarms_template_with_foreach) {
+        RRDSET *st;
+        rrdset_foreach_read(st, host) {
+            rrdset_wrlock(st);
+
```

Comment:
This will scan all the hosts charts every X seconds. Perhaps we should add a host flag to indicate that there are PENDING alarms to be configured (similarly to the dimensions being added that sets the RRDSET_FLAG_PENDING_FOREACH_ALARMS on the chart)